pkgs/development/compilers/rust/cargo.nix

```diff
@@ -16,7 +17,10 @@ rustPlatform.buildRustPackage {
   # changes hash of vendor directory otherwise
   dontUpdateAutotoolsGnuConfigScripts = true;
 
-  nativeBuildInputs = [ pkg-config cmake installShellFiles makeWrapper ];
+  nativeBuildInputs = [
+    pkg-config cmake installShellFiles makeWrapper
+    (lib.getDev pkgsHostHost.curl)
```

Comment:
curious, did you run into troubles with the normal curl as a nativeBuildInputs ?

Reply:
No, but normal curl would have been incorrect — because of callPackage splicing, it would have been the build system's version of curl, not the one we're actually linking to.  This would cause problems if e.g. http2 was enabled for the build system curl but not the host curl.

Reply:
it makes me wonder if we shouldn't change it for cargo-c as well.
(it doesn't need to be in this PR of course).

Reply:
It's unlikely to end up making much of a difference, but if you have the time it probably would be better to change it.